### PR TITLE
CppCheck fixes

### PR DESCRIPTION
**Pull request by Broxzier**
I ran CppCheck on the project to find issues and clean up the code a little bit. It's how I found https://github.com/OpenRCT2/OpenRCT2/commit/1935f795074e54ee298b0972657eba6e8f0c493f, and a few other small issues. I used the tool only for checking the files, all changes have been made by hand. Here's what I did:

- Reduced variable scopes where possible (not all, but quite some)
- Removed unneccesary operations (e.g. second condition in `while (*source == ' ' && *source != '\0')`)
- Made single-argument constructors explicit, with a few exceptions (e.g. `Nullable`)
- Newline at the end of all source and header files
- Catching exceptions by const reference instead of copies
- Removed unused variables
- Using C++ casts in C++ files
- Fixed a small issue in mapgen.c (`util_rand() % 3` -> `(util_rand() & 3)` for a random rotation)
- Added parentheses around the conditional part of ternary operator
- Ordered some initialzer lists that were out of order

I've found [one case of undefined behaviour](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/interface/Theme.cpp#L597) that I want to fix in a separate PR, because it's different from the small changes in here.

While editing some C++ files, I noticed that some use tabs, and some use 4 spaces. For those with tabs I also used tabs.

### CODE REVIEW

File: src/audio/mixer.cpp [hunk 1]
```diff
@@ -438,9 +437,11 @@ void Channel::SetGroup(int group)
 }
 
 Mixer::Mixer()
+	: deviceid(0)
+	, format({ 0 })
+	, effectbuffer(0)
+	, volume(1)
```
Comment (IntelOrca) [inline src/audio/mixer.cpp]: I am not a fan of this syntax.

File: src/world/park.c [hunk 2]
```diff
@@ -263,8 +263,8 @@ int calculate_park_rating()
 		result -= 100;
 
 		if (num_exciting_rides>0){
-			average_excitement = total_ride_excitement / num_exciting_rides;
-			average_intensity = total_ride_intensity / num_exciting_rides;
+			short average_excitement = total_ride_excitement / num_exciting_rides;
+			short average_intensity = total_ride_intensity / num_exciting_rides;
```
Comment (marijnvdwerf) [inline src/world/park.c]: Don't we have `int16` for these or something? Personally I find `char`, `short` and `word` pretty difficult to understand.

Comment (Broxzier) [inline src/world/park.c]: We _could_ replace all chars with sint8, shorts with sint16. Want me to do it for this PR?

Comment (marijnvdwerf) [inline src/world/park.c]: Not talking for everyone, but I at least would enjoy the uniformity of that. Maybe do that in another PR? Single-topic PR's are easier to check.

File: src/audio/mixer.cpp (hunk 1 above)
Comment (Broxzier) [inline src/audio/mixer.cpp]: The other initializer lists I can find are written on one line, but that's not readable for more than two members. I suggest to just place the commas at the end of the lines here, and the colon behind the method. Agreed @IntelOrca?

Comment (IntelOrca) [inline src/audio/mixer.cpp]: Fields should be initialised on the line of the definition or within the constructor body. Using this syntax makes it look like those fields are methods when they aren't. Most of the other CPP files is done the way I described.

**Comment by janisozaur** [comment]
@IntelOrca this is for .6, right?

**Comment by IntelOrca** [comment]
Yea

File: src/world/park.c (hunk 2 above)
Comment (Broxzier) [inline src/world/park.c]: I'd like to wait with that until this PR is merged, otherwise either of them will have a lot of conflicts.
